File: src/go/proxy/basbakan/GercekBasbakan.go
package basbakan

import "fmt"

type GercekBasbakan struct{
	Name string

}

func (gb *GercekBasbakan) DertDinle(dert string){
	fmt.Println("Basbakan: Dinliyorum.")
}

func (gb *GercekBasbakan) IsBul(yakinim string){
	fmt.Println("Basbakan: Bana boyle isteklerle gelmeyin.")
}

File: src/typescript/com/gokselkucuksahin/dp/structural-patterns/proxy/gercekbasbakan.ts
import {Basbakan} from "./basbakan";

export class GercekBasbakan implements Basbakan {
  dertDinle(dert: string): void {
    console.log("Basbakan: Derdinizi dinliyorum...");
  }

  isBul(dert: string): void {
    console.log("Basbakan: Bana böyle isteklerle gelmeyin.")
  }
}